virtualizarr/accessor.py

```diff
@@ -88,6 +94,43 @@ def to_icechunk(
             last_updated_at=last_updated_at,
         )
 
+    def to_xarray(self, stores: dict[str, _UpstreamObjectStore], **kwargs) -> Dataset:
+        """
+        Convert the virtual dataset to an VirtualObjectStore-backed xarray dataset which can
+        be used to load data directly without first serializing using Icechunk or Kerchunk.
+
+        Parameters
+        ----------
+        stores: dict
+            Object stores to use for fetching data. The keys should provide a prefix that can
+            be matched to urls contained within the virtual dataset (e.g., `file://`,
+            `s3://mybucket`, `s3://mybucket1`) and the values should be properly configured
+            ObjectStore instances. A store must be provided for each storage type and
+            configuration set (e.g., LocalStore, public S3Store, private S3Store).
+
+        kwargs: dict, optional
+            Additional keyword-arguments to pass to xarray.open_dataset().
+
+
+        Examples
+        --------
+
+        >>> from obstore.store import LocalStore
+        >>> stores = {"file://": LocalStore()}
+        >>> vds.virtualize.to_xarray(stores=stores)  # doctest: +SKIP
+
+        Warnings
+        --------
+        The to_xarray() functionality is experimental.
+        """
+        store = VirtualObjectStore(self.ds, stores)
+        return open_dataset(
+            store,  # type: ignore[arg-type]
+            engine="zarr",
+            backend_kwargs={"zarr_format": 3, "consolidated": False},
+            **kwargs,
+        )
```

Comment:
I think adding this `.virtualize.to_xarray` accessor method is clever, but possibly premature. I'm not entirely sure what the best API would be, but I think in the meantime we should simply expose `VirtualObjectStore`/`ManifestStore` as public (developer) API, as I sketched out in #458.

Reply:
I find this super convenient. I made it private for the time being and can remove it later in this PR if you feel strongly about making the user need to create the store and pass it to open_dataset with the necessary kwargs.